x-pack/legacy/plugins/ml/common/types/jobs.ts

```diff
@@ -63,6 +65,20 @@ export interface MlSummaryJob {
 
 export type MlSummaryJobs = MlSummaryJob[];
 
+export interface MlJobWithTimeRange extends MlJob {
```

Comment:
Agreed, this might be worth revisiting and would be good to clean up the job types. Note I created this interface as part of consuming existing JS code, the job selector uses this structure internally.